Win/PublishSkfb.h

```diff
@@ -0,0 +1,216 @@
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "stdafx.h"
+#include <string>
+#include <sstream>
+#include <Windowsx.h>
+#include "Resource.h"
+#include "ExportPrint.h"
+#include "sketchfabUploader.h"
+
+// Sketchfab API field limits
+#define SKFB_NAME_LIMIT         48
+#define SKFB_DESC_LIMIT         1024
+#define SKFB_TAG_LIMIT          39   // 48 but "mineways " is automatically added so - 9
+#define SKFB_TOKEN_LIMIT        32
+#define SKFB_PASSWORD_LIMIT     64
+
+
+static PublishSkfbData skfbPbdata;
+static std::pair<bool, std::string> lastResponse;
+static HWND uploadWindow;
+
+void getPublishSkfbData(PublishSkfbData *pEpd);
+void setPublishSkfbData(PublishSkfbData *pEpd);
+int uploadToSketchfab(HINSTANCE hInst,HWND hWnd);
+int doPublishSkfb(HINSTANCE hInst,HWND hWnd);
+INT_PTR CALLBACK managePublishWindow(HWND hDlg,UINT message,WPARAM wParam,LPARAM lParam);
+INT_PTR CALLBACK manageUploadWindow(HWND hDlg,UINT message,WPARAM wParam,LPARAM lParam);
+HANDLE uploadThreadHandle = NULL;
+
+
+void getPublishSkfbData(PublishSkfbData *pSkfbpd)
+{
+    *pSkfbpd = skfbPbdata;
+}
+
+void setPublishSkfbData(PublishSkfbData *pSkfbpd)
+{
+    skfbPbdata = *pSkfbpd;
+}
+
+int uploadToSketchfab(HINSTANCE hInst,HWND hWnd)
+{
+    // Success or failure are handled in the callback function
+    DialogBox(hInst,MAKEINTRESOURCE(IDD_UPLOAD_SKFB),hWnd, manageUploadWindow);
+    return 0;
+}
+
+
+int doPublishSkfb(HINSTANCE hInst,HWND hWnd)
+{
+    gOK = 0;
+    DialogBox(hInst,MAKEINTRESOURCE(IDD_PUBLISH_SKFB),hWnd, managePublishWindow);
+    // did we hit cancel?
+    return gOK;
+}
+
+DWORD WINAPI thread_func(LPVOID lpParameter)
+{
+	UNREFERENCED_PARAMETER(lpParameter);
+    SketchfabV2Uploader uploader;
+    lastResponse = uploader.upload(uploadWindow, skfbPbdata.skfbApiToken, skfbPbdata.skfbFilePath, skfbPbdata.skfbName, skfbPbdata.skfbDescription, skfbPbdata.skfbTags, skfbPbdata.skfbDraft, skfbPbdata.skfbPrivate, skfbPbdata.skfbPassword);
+    SendMessage(uploadWindow, SIGNAL_UPLOAD_FINISHED, 100, 0);
+    return 0;
+}
+
+INT_PTR CALLBACK manageUploadWindow(HWND hDlg,UINT message,WPARAM wParam,LPARAM lParam)
+{
+    UNREFERENCED_PARAMETER(lParam);
+
+    switch (message)
+    {
+    case WM_INITDIALOG:
+        {
+            uploadWindow = hDlg;
+            uploadThreadHandle = CreateThread(NULL, 0, thread_func, NULL, 0, 0);
+            break;
+        }
+    case SIGNAL_UPLOAD_FINISHED:
+    {
+        if(lastResponse.first){ // Upload succeeded
+            int retcode= MessageBox(NULL,
+                                    _T("Your model has been successfuly uploaded on Sketchfab.\nClick OK to open a tab on the model page"),
+                                    _T("Upload successful"),
+                                    MB_OKCANCEL | MB_ICONINFORMATION);
```

Comment:
I'm understanding that at the end of the upload we say to the user "click to see the model" or "cancel"; it seems a bit strange to me in terms of UX.